examples/flax/language-modeling/run_mlm_flax.py

```diff
@@ -792,11 +808,25 @@ def eval_step(params, batch):
         # Gather the indexes for creating the batch and do a training step
         for step, batch_idx in enumerate(tqdm(train_batch_idx, desc="Training...", position=1)):
             samples = [tokenized_datasets["train"][int(idx)] for idx in batch_idx]
-            model_inputs = data_collator(samples, pad_to_multiple_of=16)
 
+            # (GBS, seq_lenth)
+            model_inputs = data_collator(samples, pad_to_multiple_of=16)
+            step_start = time.time()
             # Model forward
+            # (# of GPU, per GPU BS, seq_length)
             model_inputs = shard(model_inputs.data)
-            state, train_metric, dropout_rngs = p_train_step(state, model_inputs, dropout_rngs)
+
+            # state, train_metric, dropout_rngs = p_train_step(state, model_inputs, dropout_rngs)
+            # Use this line for benchmark
+            state, train_metric, dropout_rngs = jax.block_until_ready(p_train_step(state, model_inputs, dropout_rngs))
```

Comment:
pmapped functions in JAX run asynchronously, so we need to call block_until_ready to make sure a particular computation has actually finished.